### PR TITLE
upgrade mysql connector J to the latest

### LINKED ISSUE #369: Maxwell crash when bootstrapping a table.
When try to bootstrap a table by inserting record into `maxwell.bootstrap` table ,Maxwell crashed with following errors:
We are using Maxwell 1.1.1
wonder if it a bug or specific to my environment?
Any help is appreciated 
-Sumo

```
15:19:46,869 ERROR SchemaScavenger - SchemaScavenger thread aborting after exception: java.sql.SQLException: Lock wait timeout exceeded; try restarting transaction
02:39:43,336 WARN  MaxwellReplicator - no heartbeat heard from server in 2003ms.  restarting replication.
02:39:43,344 INFO  TransportImpl - disconnected from mysql_host:3306
02:39:43,345 INFO  TransportImpl - connecting to host: mysql_host, port: 3306
02:39:47,826 INFO  TransportImpl - connected to host: mysql_host, port: 3306, context: AbstractTransport.Context[threadId=24418,scramble=\i{MN}MFWes4E>H.Lt%U,protocolVersion=10,serverHost=mysql_host,serverPort=3306,serverStatus=2,serverCollation=8,serverVersion=5.6.24-72.2-25.11-log,serverCapabilities=65535]
02:39:47,826 INFO  AuthenticatorImpl - start to login, user: maxwell_user, host: mysql_host, port: 3306
02:39:47,840 INFO  AuthenticatorImpl - login successfully, user: maxwell_user, detail: OKPacket[packetMarker=0,affectedRows=0,insertId=0,serverStatus=2,warningCount=0,message=<null>]
com.google.code.or.net.TransportException: Lock wait timeout exceeded; try restarting transaction
        at com.google.code.or.net.impl.Query.getFirst(Query.java:40)
        at com.google.code.or.OpenReplicator.setupChecksumState(OpenReplicator.java:268)
        at com.google.code.or.OpenReplicator.start(OpenReplicator.java:109)
        at com.zendesk.maxwell.MaxwellReplicator.ensureReplicatorThread(MaxwellReplicator.java:94)
        at com.zendesk.maxwell.MaxwellReplicator.getRow(MaxwellReplicator.java:280)
        at com.zendesk.maxwell.MaxwellReplicator.work(MaxwellReplicator.java:115)
        at com.zendesk.maxwell.RunLoopProcess.runLoop(RunLoopProcess.java:31)
        at com.zendesk.maxwell.Maxwell.run(Maxwell.java:101)
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:107)
02:39:54,257 ERROR SchemaPosition - received SQLException while trying to save to maxwell.positions:
02:39:54,257 ERROR SchemaPosition - WSREP has not yet prepared node for application use
```

### Other bootstrap error:

```
11:42:19,082 INFO  AuthenticatorImpl - start to login, user: maxwell_user, host: mysql_host, port: 3306
11:42:19,086 INFO  AuthenticatorImpl - login successfully, user: maxwell_user, detail: OKPacket[packetMarker=0,affectedRows=0,insertId=0,serverStatus=2,warningCount=0,message=<null>]
11:52:19,151 WARN  Selector - Error in I/O with dbsrs1496/10.207.146.57
12:58:10,252 INFO  MaxwellReplicator - bootstrapping started for mydb.user_tab, binlog position is BinlogPosition[binarylog.042355:33264]
java.sql.SQLException: Cannot convert value '2016-06-13 05:56:59.000' from column 4 to TIMESTAMP.
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1055)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:956)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:926)
        at com.mysql.jdbc.ResultSetRow.getTimestampFast(ResultSetRow.java:1321)
        at com.mysql.jdbc.BufferRow.getTimestampFast(BufferRow.java:573)
        at com.mysql.jdbc.ResultSetImpl.getTimestampInternal(ResultSetImpl.java:6617)
        at com.mysql.jdbc.ResultSetImpl.getTimestamp(ResultSetImpl.java:5943)
        at com.mysql.jdbc.ResultSetImpl.getObject(ResultSetImpl.java:4901)
        at com.zendesk.maxwell.bootstrap.SynchronousBootstrapper.setRowValues(SynchronousBootstrapper.java:221)
        at com.zendesk.maxwell.bootstrap.SynchronousBootstrapper.startBootstrap(SynchronousBootstrapper.java:58)
        at com.zendesk.maxwell.bootstrap.AsynchronousBootstrapper$1.run(AsynchronousBootstrapper.java:70)
        at java.lang.Thread.run(Thread.java:745)
ADMINISTRATOR0016-06-13 05:56:59.000hts/jsp/security/systemMenu.jsp2016-06-13 05:56:59.000
        at com.mysql.jdbc.StringUtils.getInt(StringUtils.java:756)
        at com.mysql.jdbc.ResultSetRow.getTimestampFast(ResultSetRow.java:1106)
        ... 8 more
12:58:10,388 INFO  TransportImpl - disconnected from mysql_host:3306
12:58:10,397 INFO  SchemaPosition - Storing final position: BinlogPosition[binarylog.042355:33264]
```

### Other bootstrap error:

```
13:09:30,344 INFO  MaxwellReplicator - bootstrapping started for mydb.user_tab, binlog position is BinlogPosition[binarylog.042356:22106]
java.sql.SQLException: Cannot convert value '2016-06-13 05:56:59.000' from column 4 to TIMESTAMP.
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1055)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:956)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:926)
        at com.mysql.jdbc.ResultSetRow.getTimestampFast(ResultSetRow.java:1321)
        at com.mysql.jdbc.BufferRow.getTimestampFast(BufferRow.java:573)
        at com.mysql.jdbc.ResultSetImpl.getTimestampInternal(ResultSetImpl.java:6617)
        at com.mysql.jdbc.ResultSetImpl.getTimestamp(ResultSetImpl.java:5943)
        at com.mysql.jdbc.ResultSetImpl.getObject(ResultSetImpl.java:4901)
        at com.zendesk.maxwell.bootstrap.SynchronousBootstrapper.setRowValues(SynchronousBootstrapper.java:221)
        at com.zendesk.maxwell.bootstrap.SynchronousBootstrapper.startBootstrap(SynchronousBootstrapper.java:58)
        at com.zendesk.maxwell.bootstrap.AsynchronousBootstrapper$1.run(AsynchronousBootstrapper.java:70)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NumberFormatException: ^MADMINISTRATOR^ZADMINISTRATOR - God Rights^\/jsp/security/systemMenu.jsp^W2016-06-13 05:56:59.000^MADMINISTRATOR^W2016-06-13 05:56:59.000^MADMINISTRATOR^A0^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
        at com.mysql.jdbc.StringUtils.getInt(StringUtils.java:756)
        at com.mysql.jdbc.ResultSetRow.getTimestampFast(ResultSetRow.java:1106)
        ... 8 more
13:09:30,459 INFO  TransportImpl - disconnected from dbsrs0879:3306
13:09:30,469 INFO  SchemaPosition - Storing final position: BinlogPosition[binarylog.042356:22106]
```

### PR DESCRIPTION
**Pull request by osheroff**
this should fix #369

### CODE REVIEW

**Comment by osheroff** [comment]
ping here, would like to get a patch release out

**Comment by osheroff** [comment]
oops @zendesk/rules

File: src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java [hunk 1]
```diff
@@ -161,7 +164,7 @@ private Long calculateRowCount(Connection connection, String db, String table) t
 	private long insertBootstrapStartRow(Connection connection, String db, String table, Long totalRows) throws SQLException {
 		LOGGER.info("inserting bootstrap start row");
 		String sql = "insert into `bootstrap` (database_name, table_name, total_rows) values(?, ?, ?)";
-		PreparedStatement preparedStatement = connection.prepareStatement(sql);
+		PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java]: so now you need to be explicit about returning generated keys?

**Comment by dadah89** [comment]
lgtm 👍

File: src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java]: yeah, I guess there was some slop in the old code that allowed it to work without

**Pull request merged**